app/src/test/java/com/codesoom/assignment/product/domain/ProductRepositoryTest.java

```diff
@@ -0,0 +1,151 @@
+            void prepare() {
+                givenProducts.add(getProductRepository().save(ProductFactory.createProduct(1L)));
+                givenProducts.add(getProductRepository().save(ProductFactory.createProduct(2L)));
+            }
+
+            @Test
+            @DisplayName("모든 장난감을 리턴한다")
+            void it_returns_all_toy() {
+                final List<Product> actualProducts = getProductRepository().findAll();
+
+                assertThat(actualProducts).hasSize(givenProducts.size());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("findById 메소드는")
+    class Describe_findById {
+        @Nested
+        @DisplayName("유효한 ID가 주어지면")
+        class Context_with_existed_id extends JpaTest {
+            private Product givenProduct;
+
+            @BeforeEach
+            void prepare() {
+                givenProduct = getProductRepository().save(ProductFactory.createProduct());
+            }
+
+            @Test
+            @DisplayName("고양이 장난감을 찾아서 리턴한다")
+            void it_returns_searched_cat_toy() {
+                final Product actualProduct = getProductRepository().findById(givenProduct.getId()).orElse(null);
+
+                assertThat(actualProduct.getName()).isEqualTo(givenProduct.getName());
+                assertThat(actualProduct.getId()).isEqualTo(givenProduct.getId());
+                assertThat(actualProduct.getPrice()).isEqualTo(givenProduct.getPrice());
+            }
+        }
+
+        @Nested
+        @DisplayName("유효하지 않은 ID가 주어지면")
+        class Context_with_non_existed_id extends JpaTest {
+            @Test
+            @DisplayName("Optional.empty()를 리턴한다")
+            void it_returns_optional_empty() {
+                assertThat(getProductRepository().findById(100L)).isEmpty();
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("save 메소드는")
+    class Describe_save {
+        @Nested
+        @DisplayName("새로운 고양이 장난감이 주어지면")
+        class Context_with_new_cat_toy extends JpaTest {
+            private final Product givenProduct = ProductFactory.createProduct();
+
+            @Test
+            @DisplayName("DB에 등록하고 등록된 장난감을 리턴한다")
+            void it_returns_registered_cat_toy() {
+                final Product actualProduct = getProductRepository().save(givenProduct);
+
+                assertThat(actualProduct.getName()).isEqualTo(givenProduct.getName());
+                assertThat(actualProduct.getPrice()).isEqualTo(givenProduct.getPrice());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("delete 메소드는")
+    class Describe_delete {
+        @Nested
+        @DisplayName("유효한 ID가 주어지면")
+        class Context_with_existed_id extends JpaTest {
+            private Product givenProduct;
+
+            @BeforeEach
+            void prepare() {
+                givenProduct = getProductRepository().save(ProductFactory.createProduct());
+            }
+
+            @Test
+            @DisplayName("DB에서 삭제한다")
```

Comment:
DB가 아닌 곳에 데이터가 저장될 수도 있습니다. Repository는 데이터 저장소를 추상화하는 레이어라는 점을 염두에 두어야 합니다.

```suggestion
            @DisplayName("삭제한다")
```